Tests/PackageLoadingTests/JSONSerializationTests.swift

```diff
@@ -54,7 +54,20 @@ class JSONSerializationTests: XCTestCase {
         let t1 = Target(name: "One")
         let t2 = Target(name: "Two", dependencies: [.Target(name: "One")])
         let package = Package(name: "Targets", targets: [t1, t2])
-        assertEqual(package: package, expected: "{\"dependencies\": [], \"exclude\": [], \"name\": \"Targets\", \"targets\": [{\"dependencies\": [], \"name\": \"One\"}, {\"dependencies\": [\"One\"], \"name\": \"Two\"}]}")
+        assertEqual(package: package, expected: "{\"dependencies\": [], \"exclude\": [], \"name\": \"Targets\", \"targets\": [{\"dependencies\": [], \"name\": \"One\"}, {\"dependencies\": [{\"name\": \"One\", \"type\": \"target\"}], \"name\": \"Two\"}]}")
+
+        // Test target dependencies.
+        var dependency: Target.Dependency = .Target(name: "foo")
+        XCTAssertEqual(dependency.toJSON().toString(), "{\"name\": \"foo\", \"type\": \"target\"}")
+
+        dependency = .ByName(name: "foo")
+        XCTAssertEqual(dependency.toJSON().toString(), "{\"name\": \"foo\", \"type\": \"byname\"}")
+
+        dependency = .Product(name: "foo", package: nil)
```

Comment:
Just a small comment: perhaps it's better if this test leaves out the package parameter, since that's probably the most common use case.  Or maybe both forms (leaving it out and passing nil) should be tested.  Essentially it's just testing that the default value is nil, but that's useful.